getting-started/developers-guide.md

```diff
@@ -2,65 +2,31 @@
-- Python 2.7.x. On Linux and OS X, this is most likely provided out of the box. See [here](https://wiki.python.org/moin/BeginnersGuide/Download) for instructions.
-
-After installing, make sure that `git`, `cmake` and `python` are accessible in PATH. Technically, CMake and a system compiler may not be needed if using a precompiled toolchain, but development options may be a bit limited without them.
-
-    $ git clone https://github.com/emscripten-core/emsdk.git
-    $ cd emsdk
-    $ ./emsdk install --build=Release sdk-incoming-64bit binaryen-master-64bit
-    $ ./emsdk activate --build=Release sdk-incoming-64bit binaryen-master-64bit
-
-## Environment setup after installation (non Windows 7 installs)
-After installing or compiling the SDK, the installation is complete. To enter an Emscripten compiler environment in the current command line prompt after downloading a precompiled toolchain or building your own, type
-
-    $ source ./emsdk_env.sh --build=Release
-
-This command adds relevant environment variables and directory entries to PATH to set up the current terminal for easy access to the compiler tools.
-
-## Compile and run a simple program
-We now have a full toolchain we can use to compile a simple program to WebAssembly. There are a few remaining caveats, however:
-
-- If we want Emscripten to generate an HTML page that runs our program, in addition to the Wasm binary and JavaScript wrapper, we have to specify an output filename with a `.html` extension.
-- Finally, to actually run the program, we cannot simply open the HTML file in a web browser because cross-origin requests are not supported for the `file` protocol scheme. We have to actually serve the output files over HTTP.
-
-The commands below will create a simple "hello world" program and compile it. The compilation step is highlighted in bold.
-
-<pre>
-$ mkdir hello
-$ cd hello
-$ cat << EOF > hello.c
-#include &lt;stdio.h&gt;
-int main(int argc, char ** argv) {
-  printf("Hello, world!\n");
-}
-EOF
-$ <b>emcc hello.c -o hello.html</b>
-</pre>
-
-To serve the compiled files over HTTP, we can use the `emrun` web server provided with the Emscripten SDK:
-
-    $ emrun --no_browser --port 8080 .
-
-Once the HTTP server is running, you can <a href="http://localhost:8080/hello.html" target="_blank">open it in your browser</a>. If you see "Hello, world!" printed to the Emscripten console, then congratulations! You've successfully compiled to WebAssembly!
+# I want to...
+
+- Compile a WebAssembly module from...
+  - C/C++
+    - [starting from scratch](https://developer.mozilla.org/en-US/docs/WebAssembly/C_to_wasm)
+    - [library that I want to port to the Web](https://developer.mozilla.org/en-US/docs/WebAssembly/existing_C_to_wasm)
```

Comment:
These 2 links are very good but I'd recommend also having the Emscripten tutorial, https://emscripten.org/docs/getting_started/Tutorial.html

Reply:
That's something I wanted to discuss separately (or I think mentioned already before?) - I believe that emscripten.org, especially the tutorial, should be updated before we can link out to it from this page.

It's what I wanted to link to initially, but, unfortunately, I think that tutorial is rather advanced and might be overwhelming for beginners in its current state with the jumps right away from "hello world" to SDL usage and more.

In this spirit, MDN links seem more tailored for a gettin started guide, as it focuses on most basic concepts (compiling, running and exposing a function) without going into advanced stuff.

Reply:
Oh, I see. Happy to discuss that separately if you don't want the noise in this PR. Also I'd be very happy to get suggestions and help for improving that page!

One reason I think an `emscripten.org` link is a good idea is that we update it together with the rest of our docs with all of emscripten. Do we know who updates those MDN links?

Reply:
I think they're maintained by community in general - since MDN is a wiki, it's easy for anyone to contribute.

This can be seen as both pros and cons at the same time, but we want to move webassembly.org to be a more community-oriented website, so maybe it's a good thing?

Anyway, happy to discuss separately as well, and, as mentioned in PR description, we can always extend / update the list, for now I just want to have a best-effort attempt at a list of getting started guides that beginners can jump into. When emscripten.org updates, it should be easy to switch over.